### ISSUE TITLE
Selective import

### ISSUE THREAD

**Issue by TimidScript** [violentmonkey/violentmonkey#1414]
1) I am not sure if this is possible, but auto-backing of scripts and values at fixed times. If one needs a reason, I recently was forced to do a fresh install of my system, and I forgot to copy some stuff before formatting. 

2) Import selection: ability to select which scripts+values to import

**Comment by Cyberavater** [comment]
Maybe related to #48.

**Comment by filbo** [comment]
> I don't think so. Remembering to back up stuff, especially before reinstalling something, becomes second nature after one painful lesson. I had one ~20 years ago and since then I use an automatic daily backup of the entire system partition. We can't expect every program to remind us to back up things just because people in general underestimate the importance of doing a regular backup of everything they don't want to lose.

This is sort of specious for deeply stored domain-specific data.  Sure you can back up the whole system; and you can restore the whole system and blow away everything.  Or you can restore it into a VM or something.  But if you want to pick out of it something specific like 'script-saved data for script xyz in violentmonkey in my abc firefox profile', it's a hairy ugly mess.  You either have to decode the storage formats of the various containers (FF storage for extensions, then VM storage for scripts); or actually _run_ the backed up system in a sufficiently 'real' manner that you can interact with those layers and manually save what was lost.

Far better to be able to configure FF-abc-TM to sync xyz's data to 'the cloud' so it is always up to date wherever I go.

**Comment by tophf** [comment]
No, the reason for auto-backup given here was specifically to restore the system so a full backup of the profile is a better solution.

**Comment by filbo** [comment]
> I recently was forced to do a fresh install of my system

This does not imply a whole-system restore, but rather a manual reconstruction using parts pulled from old saves.  If it was a whole-system restore then it *would* have restored the browser profile, therefore the extensions and their data.

While we are reading things into the original request, I read 'auto-back this stuff up to a cloud place', not 'to the local system'.  You are right that backing up locally in a separately accessible manner would invoke the browser download dialog and be super irritating.  Backing up to a cloud safety-hole should be 100% automatable.

... So I just went into VM settings > Sync  -- turned on sync to Google Drive.  Which Did Something.  I'm really not sure what.  I can't find anything stored in Google Drive.  Did it back up my VM scripts and their data?  (I set [x] Sync script status -- but what is meant by 'status'?)  Did it just back up a list of script names and whether they're enabled, and VM settings?  It's 98% opaque.

![vm-sync-me](https://user-images.githubusercontent.com/3022180/179325051-e8034e27-d46f-4cb3-8bab-c1bec5aa20ab.png)

**Comment by tophf** [comment]
Cloud sync already works automatically. As for script data, it's #48, not yet implemented.

**Comment by filbo** [comment]
Where can I _find_ my VM stuff on Google Drive?

The dialog which offers 'Sync script status' is not informative.  You're saying it syncs not just metadata, but the actual script code itself?  But not each script's GM_setValue data -- I know that much...

**Comment by tophf** [comment]
I don't know, it's probably somewhere in account settings, there should be tutorials on how to find the storage for extensions in google drive. I'm deleting the sync-related comments because it's not what this issue is about.

**Comment by tophf** [comment]
When #48 is implemented, you'll use `Sync` option for auto-backup, so this issue is about selective import.
